### PR TITLE
Fix bucket description for end of day (0:00)

### PR DESCRIPTION
**Pull request by kevinvandijk**
If the end of a period spans a full day the date will be 00:00 next day so it would be displayed wrong.
e.g. when end = "2015-05-12 00:00:00:000" we are actually displaying data until the end of May 11 so the label should say May 11, not 12

/cc @Juice10

### CODE REVIEW

**Comment by Juice10** [comment]
Your tests are failing

**Comment by kevinvandijk** [comment]
My bad, I pushed a bit too early. Fixed now.

File: src/models/period.js [hunk 1]
```diff
@@ -288,14 +289,22 @@ define(['core', './helpers/timezone-detector'], function(Openhose, timezoneDetec
...
       }
 
-      var now = Openhose.moment().tz(this.timezone());
-      var result = '';
+      // If the end spans a full day the date will be 00:00 next day so it would be displayed wrong
+      // e.g. when end = "2015-05-12 00:00:00:000" we are actually displaying data until the end of May 11
+      // so the label should say May 11, not 12. Fixing it by subtracting a day from the end in those cases
+      var endIsFullDay = parseInt(end.format('HmsSS'), 10);
+      if (!isSameday && !endIsFullDay) {
+        end.subtract(1, 'day');
```
Comment (Juice10) [inline src/models/period.js]: Isn't this doing the opposite to what that hack above it is doing?
Wouldn't it be better to just kill that?

```
if (end.millisecond() === 999) {
         end.add(1, 'millisecond');
       }
```

File: src/models/period.js [hunk 2]
```diff
@@ -288,14 +289,22 @@ define(['core', './helpers/timezone-detector'], function(Openhose, timezoneDetec
       var start = Openhose.moment(this.get('start')).tz(this.timezone());
       var end =  Openhose.moment(this.get('end')).tz(this.timezone());
 
+      var isSameday = start.format('LL') === end.format('LL');
+
       // Hack for the end being till the end of a period, not including.
       if (end.millisecond() === 999) {
         end.add(1, 'millisecond');
```
Comment (Juice10) [inline src/models/period.js]: This is the hack I'm mentioning below

File: src/models/period.js (hunk 1 above)
Comment (kevinvandijk) [inline src/models/period.js]: It's an addition to it. The above hack should eventually disappear but at the moment it works for cases when an end time of 23:59:59:999 is given, which is basically the end of the day and is passed in often at the moment and a really hairy situation to get rid off. 

So it adds 1ms to go to 00:00:00:000 the next day, after which the day is shifted back one 1 for the label. Then there's the second case where the actual right end date and time of 00:00:00:000 is passed in, where the day also has to be shifted back by 1 for the label.

Comment (Juice10) [inline src/models/period.js]: Shouldn't it just list the accurate time `23:59:59:999` on the label instead of one day before at 00:00:00:000?

**Comment by kevinvandijk** [comment]
@Juice10 Fixed up properly now, without the previously mentioned hacks, it checks for start and end of days to decide if it should display the time or not

**Comment by Juice10** [comment]
:+1: :boat:
Love it @kevinvandijk, well done!

**Comment by drakonen** [comment]
:+1: :car:

**Pull request merged**